src/serverrpc.cpp

```diff
@@ -0,0 +1,155 @@
+ * this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA
+ *
+ \******************************************************************************/
+
+#include "serverrpc.h"
+
+CServerRpc::CServerRpc ( CServer* pServer, CRpcServer* pRpcServer, QObject* parent ) : QObject ( parent )
+{
+    pRpcServer->HandleMethod ( "jamulus/getMode", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonObject result{ { "mode", "server" } };
+        response["result"] = result;
+        Q_UNUSED ( params );
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/getRecorderStatus", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonObject result{
+            { "initialised", pServer->GetRecorderInitialised() },
+            { "errorMessage", pServer->GetRecorderErrMsg() },
+            { "enabled", pServer->GetRecordingEnabled() },
+            { "recordingDirectory", pServer->GetRecordingDir() },
+        };
+
+        response["result"] = result;
+        Q_UNUSED ( params );
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/getClients", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonArray            clients;
+        CVector<CHostAddress> vecHostAddresses;
+        CVector<QString>      vecsName;
+        CVector<int>          veciJitBufNumFrames;
+        CVector<int>          veciNetwFrameSizeFact;
+
+        pServer->GetConCliParam ( vecHostAddresses, vecsName, veciJitBufNumFrames, veciNetwFrameSizeFact );
+
+        // we assume that all vectors have the same length
+        const int iNumChannels = vecHostAddresses.Size();
+
+        // fill list with connected clients
+        for ( int i = 0; i < iNumChannels; i++ )
+        {
+            if ( vecHostAddresses[i].InetAddr == QHostAddress ( static_cast<quint32> ( 0 ) ) )
+            {
+                continue;
+            }
+            QJsonObject client{
+                { "id", i },
+                { "address", vecHostAddresses[i].toString ( CHostAddress::SM_IP_PORT ) },
+                { "name", vecsName[i] },
+                { "jitterBufferSize", veciJitBufNumFrames[i] },
+                { "channels", pServer->GetClientNumAudioChannels ( i ) },
+            };
+            clients.append ( client );
+        }
+
+        // create result object
+        QJsonObject result{
+            { "clients", clients },
+        };
+        response["result"] = result;
+        Q_UNUSED ( params );
+    } );
+
+    pRpcServer->HandleMethod ( "jamulusserver/getServerProfile", [=] ( const QJsonObject& params, QJsonObject& response ) {
+        QJsonObject result{
+            { "name", pServer->GetServerName() },
+            { "city", pServer->GetServerCity() },
+            { "countryId", pServer->GetServerCountry() },
+            { "welcomeMessage", pServer->GetWelcomeMessage() },
+            { "registrationStatus", pServer->GetSvrRegStatus() },
```

Comment:
I think this should not return Jamulus implementation details (enum), but should be serialized similar to `SerializeSkillLevel`. Will work on this later.

Reply:
Done (but untested).